lib/fastlane/actions/update_project_provisioning.rb

```diff
@@ -0,0 +1,91 @@
+        profile = File.read(params[:profile])
+        p7 = OpenSSL::PKCS7.new(profile)
+        store = OpenSSL::X509::Store.new
+        cert = OpenSSL::X509::Certificate.new(File.read(params[:certificate]))
+        store.add_cert(cert)
+        verification = p7.verify([cert], store)
+        data = Plist::parse_xml(p7.data)
+        
+        # manipulate project file
+        Helper.log.info("Updating project '#{folder}' with UUID")
+        project_file = XCProjectFile.new(folder)
+        project_file.project.targets.first.build_configuration_list.build_configurations.each do |build_configuration|
+          build_configuration["buildSettings"]["PROVISIONING_PROFILE"] = data["UUID"]
+          build_configuration["buildSettings"]["CODE_SIGN_RESOURCE_RULES_PATH[sdk=*]"] = "$(SDKROOT)/ResourceRules.plist"
+        end
+        project_file.save
+
+        # complete
+        Helper.log.info("Successfully updated project settings in'#{params[:xcodeproj]}'".green)
+      end
+
+      def self.description
+        "Update projects code signing settings from your profisioning profile (.mobileprovision)"
+      end
+
+      def self.details
+        "This action retrieves a provisioning profile UUID from a provisioning profile (.mobileprovision) to set up the xcode projects' code signing settings in *.xcodeproj/project.pbxproj"
+      end
+
+      def self.available_options
+        [
+          FastlaneCore::ConfigItem.new(key: :xcodeproj,
+                                       env_name: "FL_PROJECT_PROVISIONING_PROJECT_PATH",
+                                       description: "Path to your Xcode project",
+                                       optional: true,
+                                       verify_block: Proc.new do |value|
+                                        raise "Path to xcode project is invalid".red unless File.exists?(value)
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :profile,
+                                       env_name: "FL_PROJECT_PROVISIONING_PROFILE_FILE_NAME",
+                                       description: "Path to provisioning profile (.mobileprovision)",
+                                       default_value: ENV["SIGH_PROFILE_FILE_NAME"],
+                                       verify_block: Proc.new do |value|
+                                        raise "Path to provisioning profile is invalid".red unless File.exists?(value)
+                                       end),
+          FastlaneCore::ConfigItem.new(key: :certificate,
+                                       env_name: "FL_PROJECT_PROVISIONING_CERTIFICATE_PATH",
+                                       description: "Path to apple root certificate",
+                                       default_value: "AppleIncRootCertificate.cer")
+        ]
+      end
+
+      def self.author
+        "Tobias Strebitzer (tobias.strebitzer@magloft.com)"
```

Comment:
Please no emails in this field, instead use your username if that's okay for you. You can see all existing entries by running `fastlane actions`.